### PR TITLE
updated the registration to use sencha uuid as id

### PR DESCRIPTION
**Pull request by cman131**
This fixes the issue with the id not being retained or unique to the devices: <a href='https://cycologists.fogbugz.com/f/cases/27/Groups-are-lost-when-the-application-is-restarted'>27</a>

Blocked temp files: <a href='https://cycologists.fogbugz.com/f/cases/37/Remove-swap-files-from-repos-add-gitignore'>37</a>

### CODE REVIEW

File: app/controller/Home.js [hunk 1]
```diff
@@ -84,13 +84,11 @@ Ext.define('DevCycleMobile.controller.Home', {
 					params: {
 						os: Ext.os.name + " " + Ext.os.version,
 						device: Ext.os.name,
-						tourId: this.tourInfo.data.tour_id
+						tourId: this.tourInfo.data.tour_id,
+                        id: rider_id
```
Comment (bbesmanoff) [inline app/controller/Home.js]: NP: Whitespace

File: app/controller/Home.js [hunk 2]
```diff
@@ -141,12 +139,11 @@ Ext.define('DevCycleMobile.controller.Home', {
 							params: {
 								os: Ext.os.name + " " + Ext.os.version,
 								device: Ext.os.name,
-								tourId: this.tourInfo.data.tour_id
+								tourId: this.tourInfo.data.tour_id,
+                                id: rider_id
```
Comment (bbesmanoff) [inline app/controller/Home.js]: NP: Whitespace

File: app/controller/Home.js [hunk 3]
```diff
@@ -141,12 +139,11 @@ Ext.define('DevCycleMobile.controller.Home', {
...
 							},
 							success: function(response){
 								var riderInfo = self.riderStore.first();
-								var decodedResponse = Ext.JSON.decode(response.responseText);
-								rider_id = decodedResponse.rider_id;
```
Comment (bbesmanoff) [inline app/controller/Home.js]: Why is this removed?

File: app/controller/Home.js [hunk 4]
```diff
@@ -84,13 +84,11 @@ Ext.define('DevCycleMobile.controller.Home', {
...
 						os: Ext.os.name + " " + Ext.os.version,
 						device: Ext.os.name,
-						tourId: this.tourInfo.data.tour_id
+						tourId: this.tourInfo.data.tour_id,
+                        id: rider_id
 					},
 					success: function(response){
 
-						var decodedResponse = Ext.JSON.decode(response.responseText);
-						rider_id = decodedResponse.rider_id;
```
Comment (bbesmanoff) [inline app/controller/Home.js]: Same as below

**Comment by CShapleigh** [comment]
Okay so, I know for a fact this file has:
1. Inconsistent formatting
2. Duplicated code
3. Commented out code


In my version of this fix (which is unused), I went through and fixed them out of habit. Should we include these fixes in pulls?

File: app/controller/Home.js (hunk 3 above)
Comment (CShapleigh) [inline app/controller/Home.js]: Because, we no longer set the rider_id to whatever the server shoots back at us. We now are setting the rider ID to the Ext.device.Device.uuid.

File: app/controller/Home.js (hunk 4 above)
Comment (CShapleigh) [inline app/controller/Home.js]: See above

File: app/controller/Home.js (hunk 3 above)
Comment (bbesmanoff) [inline app/controller/Home.js]: Makes sense.  Missed that the `rider_id` was declared elsewhere

**Comment by bbesmanoff** [comment]
Point (1) should definitely be included as a fix.

I also would be interested in seeing points (2) and (3) for the broken windows fixes.  The flip side of that is that we might be setting the precedent that any file we touch should be examined for broken windows.  We may want to discuss that offline prior to including the changes.

**Comment by cman131** [comment]
Styling done :+1:

**Comment by CShapleigh** [comment]
Until we discuss points 2 and 3, I like this. However, I don't think it should be merged until points 2 and 3 are addressed.

File: app/controller/Home.js [hunk 5]
```diff
@@ -11,315 +11,288 @@
...
-		if(this.timerStart == false)
-		{
-			this.timerStart = true;
-			var runner = new Ext.util.TaskRunner();
-        	var task = runner.start(this.updateGroupLocationTask);
+          } else {
+            alert(result[0].message);
+          }
+        } else {
+          alert('Could not reach the server. Please check your connection');
```
Comment (thenaterhood) [inline app/controller/Home.js]: Could you add a period after "Please check your connection"?

**Comment by thenaterhood** [comment]
I think points two and three would be worth doing, especially if they're already mostly done somewhere. We don't need to make it a blanket requirement for all the files we touch; we can examine it on a case-by-case basis and decide if it should be done.

**Comment by bbesmanoff** [comment]
The styling of this file is now spaces.  Other files appear to be tabs.  We should either come up with a standard and match all files to it or adopt theirs.  This is another offline discussion we should have

**Comment by CShapleigh** [comment]
Well I mean I like it

**Comment by thenaterhood** [comment]
Looks good to me. :european_post_office:
